### PR TITLE
Add Fastlane for automatic release deployments

### PR DESCRIPTION
**Pull request by chadyj**
### Summary:

[comment]: # (Summarise the problem and how the pull request solves it)

Fixes https://github.com/status-im/status-react/issues/3685

This PR adds Fastlane with two build lanes for iOS and Android for uploading to Play/iTC. It also adds deploy steps to Jenkins for auto release deployments.

### Install Ruby

* Recommended to install using rbenv. Instructions for [mac](https://github.com/rbenv/rbenv#homebrew-on-macos) or [git](https://github.com/rbenv/rbenv#basic-github-checkout).
* Install ruby 2.5.0 with`rbenv install 2.5.0` (or check the [ruby releases](https://www.ruby-lang.org/en/downloads/releases/) for new versions.)
* Set this ruby version globally with`rbenv global 2.5.0` or locally in the repo with `rbenv local 2.5.0`.
* Install bundler to manage ruby gems `gem install bundler`

### Install Fastlane

In the project root run `bundle install`.  This can also be done as a first build step in CI.

### Android Setup

To setup Play credentials grab the json_key_file from Lastpass and create an environmental variable `export SUPPLY_JSON_KEY_DATA='{JSON contents NO line breaks}'`.

### iOS Setup

Add a `APPLE_ID` environmental variable for the iTC account. There is a sandboxed account in Lastpass.

To setup iTC [credentials](https://docs.fastlane.tools/advanced/#adding-credentials) run `bundle exec fastlane fastlane-credentials add --username name@email.com`.

### Slack Setup

To have Fastlane send Slack notifications of successful deploys add a `SLACK_URL` environmental variable.

# Upload builds

To upload an APK to the Play store run `bundle exec fastlane android nightly`. Fastlane will upload the APK from `android/app/build/outputs/apk/release/app-release.apk`.

To upload an IPA to iTunes Connect run `bundle exec fastlane deploy`.  Fastlane will upload the IPA from `status/StatusIm.ipa`.

### CODE REVIEW

File: fastlane/Fastfile [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+platform :android do
+  desc "Deploy a new alpha build to Google Play"
+  lane :deploy do
+    upload_to_play_store(
+      track: "alpha",
+      apk: "android/app/build/outputs/apk/release/app-release.apk"
+    )
+    slack(message: "New build uploaded to Google Play")
+  end
+end
```
Comment (andytudhope) [inline fastlane/Fastfile]: Have you thought about adding
```
lane :clean do
  clean_testflight_testers(dry_run: true, days_of_inactivity: 60)
  slack(message: "Cleaned TestFlight users")
end
```
Or something similar here? Not necessary, but nice for completion.

File: fastlane/README.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+fastlane documentation
```
Comment (andytudhope) [inline fastlane/README.md]: Is this file really necessary?

File: fastlane/Fastfile (hunk 1 above)
Comment (chadyj) [reply fastlane/Fastfile]: That's a nice touch. There is so much more I wanted to add but thought it best for a future change.

File: fastlane/README.md (hunk 2 above)
Comment (chadyj) [reply fastlane/README.md]: Not really. I kept it around since I saw other projects keeping it.
I can add it to .gitignore.

**Review by andytudhope — APPROVED** [review]
LGTM

**Review by jeluard — APPROVED** [review]
